odoo/addons/base/models/res_partner.py

```diff
@@ -443,7 +439,7 @@ def _compute_same_vat_partner_id(self):
             # so that you can reactivate it instead of creating a new one, which would lose its history.
             Partner = self.with_context(active_test=False).sudo()
             vats = [partner.vat]
-            should_check_vat = partner.vat and len(partner.vat) != 1
+            should_check_vat = partner.vat and partner.vat not in ['/', 'na', 'NA']
```

Comment:
Would make a small tool to re-use in base_vat, so that if we add some terms / change our mind, we don't have to grep "NA" in the code :)